embassy-boot/boot/src/lib.rs

```diff
@@ -599,7 +643,13 @@ impl FirmwareUpdater {
         Ok(())
     }
 
-    // Write to a region of the DFU page
+    /// Write data to a flash page.
+    ///
+    /// The buffer must follow alignment requirements of the target flash and a multiple of page size big.
+    ///
+    /// # Safety
+    ///
+    /// Failing to meet alignment and size requirements may result in a panic.
     pub async fn write_firmware<F: AsyncNorFlash>(
```

Comment:
Unrelated to this refactor - should the user be able to write to the DFU partition when the state is SWAP? I added precondition for it into my FirmwareUpdater, because then in the case of a reboot, the firmware didn't boot.

Reply:
Yup, maybe that's a good check. I'll add it.